p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -201,7 +201,7 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metri
             }
 
             // Attempt to reserve a slot for the handshake
-            let Some(reservation) = self.handshake_limiter.try_acquire() else {
+            let Some(reservation) = self.handshake_limiter.try_acquire(ip) else {
```

Comment:
We now ensure there is at most 1 concurrent handshake per IP!

Reply:
This may make local network examples run slightly slower (all share same IP) but doubt it'll be that bad.